Makefile

```diff
@@ -0,0 +1,20 @@
+golangci-lint-version = v1.56.1
+ifeq ($(GOOS),windows)
+  WORKSPACE := ${shell go env GOPATH}
+else
+  WORKSPACE := ${shell go env GOPATH | rev | cut -d: -f1 | rev}
+endif
+
+.PHONY: install-golangci-lint
+install-golangci-lint:
+	$(call print-target)
+	curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh >lint-install.sh
+	chmod u+x ./lint-install.sh && ./lint-install.sh -b $(WORKSPACE)/bin $(golangci-lint-version)
+	$(RM) ./lint-install.sh
+
+.PHONY: ci-lint
+ci-lint: install-golangci-lint
+	$(call print-target)
+	GOOS=linux "$(WORKSPACE)/bin/golangci-lint" run --timeout 2m
+	GOOS=windows "$(WORKSPACE)/bin/golangci-lint" run --timeout 2m
+	GOOS=darwin "$(WORKSPACE)/bin/golangci-lint" run --timeout 2m
```

Comment:
I'm curious how GOOS affects the linter? I'd have hoped it would just statically analyze everything, not skip stuff for other OSes.

Reply:
If a file has a build tag that is not satisfied by the current environment then it's skipped by the linter.